src/main/starlark/builtins_bzl/common/objc/transitions.bzl

```diff
@@ -17,75 +17,12 @@
 transition = _builtins.toplevel.transition
 
 def _cpu_string(platform_type, settings):
-    arch = _determine_single_architecture(platform_type, settings)
-    if platform_type == MACOS:
-        return "darwin_{}".format(arch)
-
-    return "{}_{}".format(platform_type, arch)
-
-def _determine_single_architecture(platform_type, settings):
     apple_split_cpu = settings["//command_line_option:apple_split_cpu"]
-    if apple_split_cpu != None and len(apple_split_cpu) > 0:
-        return apple_split_cpu
-    if platform_type == IOS:
-        ios_cpus = settings["//command_line_option:ios_multi_cpus"]
-        if len(ios_cpus) > 0:
-            return ios_cpus[0]
-        cpu_value = settings["//command_line_option:cpu"]
-        if cpu_value.startswith(IOS_CPU_PREFIX):
-            return cpu_value[len(IOS_CPU_PREFIX):]
-        if cpu_value == "darwin_arm64":
-            return "sim_arm64"
-        return DEFAULT_IOS_CPU
-    if platform_type == VISIONOS:
-        cpus = settings["//command_line_option:visionos_cpus"]
-        if len(cpus) > 0:
-            return cpus[0]
-        cpu_value = settings["//command_line_option:cpu"]
-        if cpu_value == "darwin_arm64":
-            return "sim_arm64"
-        return DEFAULT_VISIONOS_CPU
-    if platform_type == WATCHOS:
-        watchos_cpus = settings["//command_line_option:watchos_cpus"]
-        if len(watchos_cpus) == 0:
-            return DEFAULT_WATCHOS_CPU
-        return watchos_cpus[0]
-    if platform_type == TVOS:
-        tvos_cpus = settings["//command_line_option:tvos_cpus"]
-        if len(tvos_cpus) == 0:
-            return DEFAULT_TVOS_CPU
-        return tvos_cpus[0]
-    if platform_type == MACOS:
-        macos_cpus = settings["//command_line_option:macos_cpus"]
-        if macos_cpus:
-            return macos_cpus[0]
-        cpu_value = settings["//command_line_option:cpu"]
-        if cpu_value.startswith(DARWIN_CPU_PREFIX):
-            return cpu_value[len(DARWIN_CPU_PREFIX):]
-        return DEFAULT_MACOS_CPU
-    if platform_type == CATALYST:
-        catalyst_cpus = settings["//command_line_option:catalyst_cpus"]
-        if len(catalyst_cpus) == 0:
-            return DEFAULT_CATALYST_CPU
-        return catalyst_cpus[0]
-
-    fail("ERROR: Unhandled platform type {}".format(platform_type))
-
-IOS = "ios"
-VISIONOS = "visionos"
-WATCHOS = "watchos"
-TVOS = "tvos"
-MACOS = "macos"
-CATALYST = "catalyst"
-IOS_CPU_PREFIX = "ios_"
-VISIONOS_CPU_PREFIX = "visionos_"
-DARWIN_CPU_PREFIX = "darwin_"
-DEFAULT_IOS_CPU = "x86_64"
-DEFAULT_VISIONOS_CPU = "x86_64"
-DEFAULT_WATCHOS_CPU = "i386"
-DEFAULT_TVOS_CPU = "x86_64"
-DEFAULT_MACOS_CPU = "x86_64"
-DEFAULT_CATALYST_CPU = "x86_64"
+    if apple_split_cpu:
+        if platform_type == "macos":
+            return "darwin_{}".format(apple_split_cpu)
+        return "{}_{}".format(platform_type, apple_split_cpu)
+    return settings["//command_line_option:cpu"]
```

Comment:
erm wouldn't removing all the other platforms here mean you couldn't build an objc library directly for iOS anymore? I feel like this change would likely be about the same as removing the transition all together?

Reply:
(Probably best to look at https://github.com/bazelbuild/bazel/pull/19235#discussion_r1294070371 first, just to make sure we're on the same page. Thanks for thinking about it with me!)

To clarify: Not removing all the others; instead unifying a separate condition. (Note that despite how the diff preview decides to align, I've moved this code down from above, not replaced all the platform-specific cases with that macOS if statement. It's all the platform-specific, non-fat-binary-because-no-rules_apple-binary fallback cases that are unified into the last line, looking at CPU. I realize I still need to make the case to you that that's the right, consistent analysis, hence attempting to order the [other comment](https://github.com/bazelbuild/bazel/pull/19235#discussion_r1294070371) first.)

To build an objc_library directly for iOS (rather than the default, Mac) one would now just specify --cpu (as elsewhere). Previously, (I think?) one couldn't build an objc_library directly for iOS with public APIs, since --apple_platform_type is undocumented (and explicitly labeled as not for direct use), and you'd have to specify it to hit the iOS case when building objc_library directly.